packages/accounts/src/light-account/utils.ts

```diff
@@ -91,3 +92,35 @@ export const LightAccountUnsupported1271Impls = [
 export const LightAccountUnsupported1271Factories = new Set(
   LightAccountUnsupported1271Impls.map((x) => x.factoryAddress)
 );
+
+export const getLightAccountVersion = async <A extends SmartContractAccount>(
```

Comment:
this probably shouldn't be generic on `SmartContractAccount` but on `LightAccount`

Reply:
then typescript errors because it cannot detect `getImplementationAddress` and `getFactoryAddress` methods on the account.